### PR TITLE
fix: allow positive i32 number for dimension priority in dimension create

### PR DESCRIPTION
**Pull request by ayushjain17**
## Problem
DB supports i32 values for priority but endpoints were restricted to u16

## Solution
allow positive i32 number for dimension priority in dimension create

### CODE REVIEW

File: crates/context_aware_config/src/api/dimension/types.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use serde_json::Value;
 #[derive(Debug, Deserialize)]
 pub struct CreateReq {
     pub dimension: String,
-    pub priority: u16,
+    pub priority: i32,
```
Comment (Datron) [inline crates/context_aware_config/src/api/dimension/types.rs:7]: If we are expecting only positive numbers, maybe use u32?
```suggestion
    pub priority: u32,
```

File: crates/frontend/src/components/dimension_form.rs [hunk 2]
```diff
@@ -195,8 +195,8 @@ where
                             class="input input-bordered w-full max-w-md"
                             value=priority.get()
                             on:change=move |ev| {
-                                logging::log!("{:?}", event_target_value(& ev).parse::< u16 > ());
-                                match event_target_value(&ev).parse::<u16>() {
+                                logging::log!("{:?}", event_target_value(& ev).parse::< u32 > ());
```
Comment (Datron) [inline crates/frontend/src/components/dimension_form.rs:198]:
```suggestion
                                logging::log!("{:?}", event_target_value(& ev).parse::<u32>());
```

**Review by Datron — APPROVED** [review]

File: crates/context_aware_config/src/api/dimension/types.rs (hunk 1 above)
Comment (ayushjain17) [reply crates/context_aware_config/src/api/dimension/types.rs:7]: avoided this, as we needed i32 in the next ops,
and then conversion from u32 to i32 is more likely to panic

File: crates/context_aware_config/src/api/dimension/handlers.rs [hunk 3]
```diff
@@ -37,6 +37,10 @@ async fn create(
         return Err(bad_argument!("Priority should be greater than 0"));
     }
 
+    if req.priority < 0 {
+        return Err(bad_argument!("Expected positive value for priority",));
+    }
+
```
Comment (Datron) [inline crates/context_aware_config/src/api/dimension/handlers.rs:43]: Can we merge both of these checks?

File: crates/context_aware_config/src/api/dimension/types.rs (hunk 1 above)
Comment (ShubhranshuSanjeev) [reply crates/context_aware_config/src/api/dimension/types.rs:7]: Where exactly @ayushjain17 ?

**Review by ShubhranshuSanjeev — CHANGES_REQUESTED** [review]

**Review by pratikmishra356 — APPROVED** [review]

**Review by ShubhranshuSanjeev — APPROVED** [review]

**Pull request merged**